scripts/theo/android/generate-values.ts

```diff
@@ -0,0 +1,96 @@
+import { writeFileSync } from 'fs';
+import { join } from 'path';
+import {
+  convert,
+  Format,
+  Prop,
+  registerFormat,
+  registerTransform,
+  registerValueTransform,
+} from 'theo';
+import xml from 'xml';
+import { opacity } from '../helpers';
+
+const valuesName = process.env.VALUES_NAME;
+if (!valuesName) throw new Error('No values name provided');
+
+const fileSuffix = process.env.NIGHT_MODE ? '-night' : '';
+
+const srcPath = join(__dirname, `src/values/${valuesName}${fileSuffix}.json`);
+const destPath = join(
+  __dirname,
+  `libraries/android/castor-tokens/src/main/res/values${fileSuffix}/castor-${valuesName}.xml`
+);
+
+registerValueTransform(
+  'ods/color/rgba',
+  (prop: Prop) => prop.get('type') === 'color',
+  (prop: Prop) => `rgba(${prop.get('value')}, ${opacity(prop)})`
+);
+
+registerTransform('android', [
+  'ods/color/rgba',
+  // keep default Theo transforms
+  'color/hex8rgba',
+  'relative/pixelValue',
+  'absolute/dp',
+  'percentage/float',
+]);
+
+(
+  [
+    {
+      type: 'android-color.xml',
+      propTypeFilter: 'color',
+      resourceKey: 'color',
+    },
+    {
+      type: 'android-dimen.xml',
+      propTypeFilter: 'size',
+      resourceKey: 'dimen',
+    },
+  ] as const
+).forEach(({ type, propTypeFilter, resourceKey }) => {
+  registerFormat(type, (result) => {
+    const obj = {
+      resources: result
+        .get('props')
+        .map((prop: Prop) => {
+          if (prop.get('type') !== propTypeFilter) return;
+
+          const fullName = `${prop.get('category')}_${prop.get('name')}`;
+          const attrName = 'ods_' + fullName.replace(/[-]/g, '_');
```

Comment:
💅🏻  [nitpick]

```suggestion
          const attrName = 'ods_' + fullName.replaceAll('-', '_');
```

Reply:
Had to revert this, as `esbuild` was failing:

<img width="1343" alt="Screenshot 2021-06-08 at 13 59 04" src="https://user-images.githubusercontent.com/20243687/121189126-b0e34780-c861-11eb-9542-5de608a4742c.png">

Reply:
It's not `esbuild`, it's `node` that doesn't have it yet 🤦🏻‍♂️ 

My bad, `/[-]/g` and `/-/g` are the same though, so we could simplify there at least.